src/ir/type-updating.cpp

```diff
@@ -307,17 +330,18 @@ void handleNonDefaultableLocals(Function* func, Module& wasm) {
     if (!set->isTee() || set->type == Type::unreachable) {
       continue;
     }
-    auto type = func->getLocalType(set->index);
-    if (type.isNonNullable()) {
+    if (badIndexes.count(set->index)) {
```

Comment:
It looks like we could remove the `if (!func->isVar(...` case above since it is implicit in this new check.

Reply:
Interesting idea. I think you are right that it would work 100% correctly that way, and is shorter code. OTOH it may be slower, since `isVar` is basically `index >= numParams` - faster than a set lookup (ordered or not).

Reply:
Perhaps if we used `std::vector<bool>` instead of a set then this would be fast either way. OTOH the number of set bits would be 0 in almost all cases.

Reply:
I would prefer the shorter code if we don't _strongly_ suspect (or measure) that it is faster this way. (It doesn't seem clear cut to me and my guess is that it doesn't make much of a difference).

But LGTM whichever way you decide.

Reply:
I do strongly suspect the current way is faster, just because a set lookup is fairly costly, isn't it? It's either using a tree and traversing it, or it's doing math to do hashing.

Reply:
I guess I would expect the math to be pretty fast, but `std::unordered_map` is still notoriously slow, so I guess you're probably right.